core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleParser.java

```diff
@@ -304,7 +305,7 @@ protected void parsePrefixID() throws IOException, RDFParseException, RDFHandler
 				throwEOFException();
 			}
 
-			prefixID.append(Character.toChars(c));
+			Chars.append(prefixID, c);
```

Comment:
prefixID could be a private final member variable in addition to the other StringBuilder.